templates/common_macros.html.j2

```diff
@@ -859,3 +861,170 @@
     {% endif %}
   {% endfor %}
 {% endmacro %}
+
+{%- macro render_functional_chains(chains) -%}
+  {%- for chain in chains -%}
+    <h3>{{ linked_name_with_icon(chain) }}</h3>
+    {{ chain.context_diagram }}
+    <p>{{ description(chain) | safe }}</p>
+  {%- endfor -%}
+{%- endmacro -%}
```

Comment:
This macro is used exactly once.

This doesn't need to be a macro, just inline it into `system-definition.html.j2`.